openmdao/core/component.py

```diff
@@ -1934,14 +1937,488 @@ def col_size_iter():
+            else:
+                meta_with_defaults = FiniteDifference.DEFAULT_OPTIONS.copy()
+            meta_with_defaults.update(meta)
+
+            _, wrtpats = keypats
+            # For each of the partials, check to see if the
+            #   check partials options are different than the options used to compute
+            #   the partials
+            for _, wrtvars in self._find_wrt_matches(wrtpats):
+                for var in wrtvars:
+                    # we now have individual vars like 'x'
+                    # get the options for checking partials
+                    fd_options, _ = _get_fd_options(var, requested_method, local_opts, step,
+                                                    form, step_calc, alloc_complex,
+                                                    minimum_step)
+                    # compare the compute options to the check options
+                    if fd_options['method'] != meta_with_defaults['method']:
+                        all_same = False
+                    else:
+                        all_same = True
+                        if fd_options['method'] == 'fd':
+                            option_names = ['form', 'step', 'step_calc', 'minimum_step',
+                                            'directional']
+                        else:
+                            option_names = ['step', 'directional']
+                        for name in option_names:
+                            if fd_options[name] != meta_with_defaults[name]:
+                                all_same = False
+                                break
+                    if all_same:
+                        msg = (f"Checking partials with respect to variable '{var}' in component "
+                               f"'{self.pathname}' using the same method and options as are used "
+                               "to compute the component's derivatives will not provide any "
+                               "relevant information on the accuracy.\n"
+                               "To correct this, change the options to do the "
+                               "check_partials using either:\n"
+                               "     - arguments to Problem.check_partials.\n"
+                               "     - arguments to Component.set_check_partial_options")
+
+                        issue_warning(msg, prefix=self.msginfo,
+                                      category=OMInvalidCheckDerivativesOptionsWarning)
+
+    def check_partials(self, out_stream=_DEFAULT_OUT_STREAM,
+                       compact_print=False, abs_err_tol=1e-6, rel_err_tol=1e-6,
+                       method='fd', step=None, form='forward', step_calc='abs',
+                       minimum_step=1e-12, force_dense=True, show_only_incorrect=False,
+                       show_worst=True):
+        """
+        Check partial derivatives comprehensively for all components in your model.
```

Comment:
copy/paste error -- should just check this component.

Reply:
fixed